### PR TITLE
chore: composition to add ListState<T> helpers

### PR DESCRIPTION
**Pull request by mchamberlain-mdsol**
Adding `isEmpty` and `contains` for Flink ListState. Generally, you'll want to use `MapState` for "O(1)" gets/contains, but this is in case you simply want to use a list instead (when you're searching across).

### CODE REVIEW

**Review by rbalaban-mdsol — DISMISSED** [review]
Looks good

File: src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package io.epiphanous.flinkrunner.flink.state
+
+
+class ListState[T] extends Serializable {
```
Comment (nextdude) [inline src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala:4]: What do you think about using a type class approach instead to add `isEmpty` and `contains` methods to Flink's `ListState` class. For instance, something like this (untested) code:

```scala
package io.epiphanous.flinkrunner.util

import org.apache.flink.api.common.state.ListState

import scala.collection.JavaConverters._

object StateUtils {
  implicit class RichListState[T](listState: ListState[T]) {
    def _iterable: Iterable[T]        =
      Option(listState.get()).map(_.asScala).getOrElse(Iterable.empty[T])
    def isEmpty: Boolean              = _iterable.isEmpty
    def contains(element: T): Boolean = _iterable.exists(_ == element)
  }
}
```

This should then let you use `isEmpty` and `contains` directly on instances of flink's `ListState` class.

**Review by nextdude — COMMENTED** [review]
Thanks for the contribution!

File: src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      item => {
+        if(item.equals(element)) {
+          return true
+        }
+      }
+    )
+    false
+  }
+
+  def update(elements : java.util.List[T]) = this.state.update(elements)
```
[the remark this answers is not in the document]
Comment (mchamberlain-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala:28]: That's fair, this was just matching parameters.

File: src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala (hunk 1 above)
Comment (mchamberlain-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/flink/state/ListState.scala:4]: Okay, side conversation + https://docs.scala-lang.org/overviews/core/implicit-classes.html will update.

File: src/main/scala/io/epiphanous/flinkrunner/flink/state/RichStateUtils.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package io.epiphanous.flinkrunner.flink.state
+import scala.collection.JavaConverters._
+
+object RichStateUtils {
+  implicit class RichListState[T](listState: org.apache.flink.api.common.state.ListState[T]) {
+    def _iterator: Iterator[T] = listState.get().iterator().asScala
+    def isEmpty: Boolean = _iterator.isEmpty
+    def contains(element: T): Boolean = _iterator.contains(element)
+    def find(element: T) : T = _iterator.find( v => v.equals(element)).get
```
Comment (nextdude) [inline src/main/scala/io/epiphanous/flinkrunner/flink/state/RichStateUtils.scala:9]: Can we add tests for the `find` and `length` methods?

**Review by nextdude — CHANGES_REQUESTED** [review]
Good, save for adding a couple tests. I'm trying to increase test coverage more generally in flinkrunner and this is a good opportunity not to miss some tests cause they're pretty easy.

File: src/main/scala/io/epiphanous/flinkrunner/flink/state/RichStateUtils.scala (hunk 3 above)
Comment (mchamberlain-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/flink/state/RichStateUtils.scala:9]: Of course, will do.

**Review by nextdude — APPROVED** [review]
Looks good.

**Pull request merged**